pkgs/development/python-modules/nbconvert/default.nix

```diff
@@ -30,6 +31,16 @@ buildPythonPackage rec {
     sha256 = "cbbc13a86dfbd4d1b5dee106539de0795b4db156c894c2c5dc382062bbc29002";
   };
 
+  # Add $out/share/jupyter to the list of paths that are used to search for
+  # various exporter templates
+  patches = [
+    ./templates.patch
+  ];
+
+  postPatch = ''
+    substituteAllInPlace ./nbconvert/exporters/templateexporter.py
+  '';
+
```

Comment:
I think converting it to do patches and substitute in once like here https://github.com/NixOS/nixpkgs/blob/883c55748fa03834d182aeef5336b0627095e074/pkgs/applications/audio/curseradio/default.nix#L20 would be a bit cleaner.

Reply:
I was thinking about that but I just couldn't figure out how to pass `out` that way.. Is it somehow available as a nix variable too, not just environment variable in the shells?

Reply:
`my_var = placeholder "out";`

Reply:
Won't `placeholder "out"` produce the output path of the derivation for the patch?

Reply:
If you would patch the patch in a separate derivation, then yes. Here the patching is done in the main derivation, so $out points to that of the main derivation. But right, it won't work with @SuperSandro2000 suggestion.

Reply:
Thanks, nice to know about `placeholder`! :+1: So, did I now understand correctly that I shouldn't make the change requested byt @SuperSandro2000 because it wouldn't work (because `out = placeholder "out"` would use the output path of the patched patch, not the output path of nbconvert)?

Reply:
Indeed, that's what happens. Using this:

```
  patches = [
    (substituteAll {
      src = ./templates.patch;
      out = placeholder "out";
    })
  ];
```

I get the following line in the final output:

```
        root_dirs.append(os.path.join("/nix/store/nwvxx6bc870aj89p3vkwsrm5fp6xji2b-templates.patch", "share", "jupyter"))
```

So, indeed, it's not pointing to nbconvert but the patch.